libbeat/cmd/instance/beat_test.go

```diff
@@ -269,3 +271,151 @@ func (r *outputReloaderMock) Reload(
 	r.cfg = cfg
 	return nil
 }
+
+func TestPromoteOutputQueueSettings(t *testing.T) {
+	tests := map[string]struct {
+		input                 []byte
+		memEvents             int
+		expectValidationError bool
+	}{
+		"blank": {input: []byte(""),
+			memEvents: 4096},
+		"defaults": {input: []byte(`
+name: mockbeat
+output:
+  elasticsearch:
+    hosts:
+      - "localhost:9200"
+`),
+			memEvents: 4096},
+		"topLevelQueue": {input: []byte(`
+name: mockbeat
+queue:
+  mem:
+    events: 8096
+output:
+  elasticsearch:
+    hosts:
+      - "localhost:9200"
+`),
+			memEvents: 8096},
+		"outputLevelQueue": {input: []byte(`
+name: mockbeat
+output:
+  elasticsearch:
+    hosts:
+      - "localhost:9200"
+    queue:
+      mem:
+        events: 8096
+`),
+			memEvents: 8096},
```

Comment:
Nit: Could totally be just me, but I found these test cases a bit hard to read.  I think it might've helped to have the opening and closing braces of each test case struct on their own lines so the struct fields all lined up nicely.

Reply:
See if the new fmt is better.  I'm open to any suggestions, I just want to be sure the yaml looks like yaml.